libcaf_core/caf/event_based_mail.test.cpp

```diff
@@ -499,6 +505,832 @@ TEST("send request message to an invalid receiver") {
   }
 }
 
+template <typename Fn>
+caf::actor make_server(caf::actor_system& sys, Fn fn) {
+  auto sf = [&]() -> behavior {
+    return {[&](int x, int y) { return fn(x, y); }};
+  };
+  return sys.spawn(sf);
+}
+
+TEST("send fan_out_request messages that return a result") {
+  auto [self, launch] = sys.spawn_inactive();
+  auto self_hdl = actor_cast<actor>(self);
+  std::vector<actor> workers{
+    make_server(sys, [](int x, int y) { return x + y; }),
+    make_server(sys, [](int x, int y) { return x + y; }),
+    make_server(sys, [](int x, int y) { return x + y; }),
+  };
+  dispatch_messages();
+  auto sum = std::make_shared<int>(0);
+  auto err = std::make_shared<error>();
+  SECTION("then with policy select_all") {
+    self->mail(1, 2)
+      .fan_out_request(workers, infinite, policy::select_all_tag)
+      .then([=](std::vector<int> results) {
+        for (auto result : results)
+          test::runnable::current().check_eq(result, 3);
+        *sum = std::accumulate(results.begin(), results.end(), 0);
+      });
+    launch();
```

Comment:
Using an inactive actor here works, but I think it is overly brittle here and, at least for me, made the test hard to read to understand what's going on.

I would much prefer to simply spawn the senders in-place:

```suggestion
    auto sender = sys.spawn([workers, sum](event_based_actor* self) {
      self->mail(1, 2)
        .fan_out_request(workers, infinite, policy::select_all_tag)
        .then([sum](std::vector<int> results) {
          for (auto result : results)
            test::runnable::current().check_eq(result, 3);
          *sum = std::accumulate(results.begin(), results.end(), 0);
        });
      });
```

Same for all the tests below.